test/test_THDM_python_interface.py.in

```diff
@@ -0,0 +1,378 @@
+sm = gm2calc.SM()
+sm.set_alpha_em_mz(1.0/128.94579)
+sm.set_alpha_em_0(0.00729735)
+sm.set_alpha_s_mz(0.1184)
+sm.set_mw(80.385)
+sm.set_mz(91.1876)
+sm.set_mh(125.09)
+sm.set_mu(2,173.34)
+sm.set_mu(1,1.28)
+sm.set_mu(0,0.0022)
+sm.set_md(2,4.18)
+sm.set_md(1,0.096)
+sm.set_md(0,0.0047)
+sm.set_ml(2,1.77684)
+sm.set_ml(1,0.1056583715)
+sm.set_ml(1,0.000510998928)
+sm.set_mv(2,0.)
+sm.set_mv(1,0.)
+sm.set_mv(0,0.)
+sm.set_ckm(Matrix3cd().Identity())
+
+# Setup the point information
+basis = gm2calc.thdm.Mass_basis()
+basis.yukawa_type = gm2calc.thdm.Yukawa_type.general
+basis.mh = 125.
+basis.mH = 400.
+basis.mA = 420.
+basis.mHp = 440.
+basis.sin_beta_minus_alpha = 0.999
+basis.lambda_6 = 0.2
+basis.lambda_7 = 0.1
+basis.tan_beta = 3.
+basis.m122 = 40000.
+basis.Piu = Matrix3cd().setZero()
+basis.Pid = Matrix3cd().setZero()
+basis.Pil = Matrix3cd().setZero()
+basis.Piu[4] = 0.1
+basis.Pid[4] = 0.2
+basis.Pil[4] = 0.3
+
+config = gm2calc.thdm.Config()
+config.force_output = False;
+config.running_couplings = False
+
+try:
+    model = gm2calc.THDM(basis,sm,config)
+    amu1L = gm2calc.calculate_amu_1loop(model) 
+    amu2Lf = gm2calc.calculate_amu_2loop_fermionic(model)
+    amu2Lb = gm2calc.calculate_amu_2loop_bosonic(model)
+    amu = amu1L + amu2Lf + amu2Lb
+    delta_amu = gm2calc.calculate_uncertainty_amu_2loop(model)
+except gm2calc.Error as e:
+    print(e.what)
+
+testclose(amu,1.1234764842312642*10.**(-7))
+testclose(amu1L,8.096409030210353*10.**(-10))
+testclose(delta_amu,9.141909743537708*10.**(-9))
+
+
+# THDM gauge basis point for type II
+
+# Setup the SM parameters
+sm = gm2calc.SM()
+sm.set_alpha_em_mz(1.0/128.94579)
+sm.set_alpha_em_0(0.00729735)
+sm.set_alpha_s_mz(0.1184)
+sm.set_mw(80.385)
+sm.set_mz(91.1876)
+sm.set_mh(125.09)
+sm.set_mu(2,173.34)
+sm.set_mu(1,1.28)
+sm.set_mu(0,0.0022)
+sm.set_md(2,4.18)
+sm.set_md(1,0.096)
+sm.set_md(0,0.0047)
+sm.set_ml(2,1.77684)
+sm.set_ml(1,0.1056583715)
+sm.set_ml(1,0.000510998928)
+sm.set_mv(2,0.)
+sm.set_mv(1,0.)
+sm.set_mv(0,0.)
+sm.set_ckm(Matrix3cd().Identity())
+
+# Setup the point information
+basis = gm2calc.thdm.Gauge_basis()
+basis.yukawa_type = gm2calc.thdm.Yukawa_type.type_2
+basis.lambda_1 = 0.7
+basis.lambda_2 = 0.6
+basis.lambda_3 = 0.5
+basis.lambda_4 = 0.4
+basis.lambda_5 = 0.3
+basis.lambda_6 = 0.2
+basis.lambda_7 = 0.1
+basis.tan_beta = 3.
+basis.m122 = 40000.
+basis.Piu = Matrix3cd().setZero()
+basis.Pid = Matrix3cd().setZero()
+basis.Pil = Matrix3cd().setZero()
+
+config = gm2calc.thdm.Config()
+config.force_output = False;
+config.running_couplings = False
+
+try:
+    model = gm2calc.THDM(basis,sm,config)
+    amu1L = gm2calc.calculate_amu_1loop(model) 
+    amu2Lf = gm2calc.calculate_amu_2loop_fermionic(model)
+    amu2Lb = gm2calc.calculate_amu_2loop_bosonic(model)
+    amu = amu1L + amu2Lf + amu2Lb
+    delta_amu = gm2calc.calculate_uncertainty_amu_2loop(model)
+except gm2calc.Error as e:
+    print(e.what)
+
+testclose(amu,3.2720909955357482*10.**(-11))
```

Comment:
Can you please check that these tests actually work? When I print out the value of `amu` as
~~~.patch
diff --git a/test/test_THDM_python_interface.py.in b/test/test_THDM_python_interface.py.in
index bb2790b..4c99585 100644
--- a/test/test_THDM_python_interface.py.in
+++ b/test/test_THDM_python_interface.py.in
@@ -366,6 +366,7 @@ try:
     amu2Lb = gm2calc.calculate_amu_2loop_bosonic(model)
     amu = amu1L + amu2Lf + amu2Lb
     delta_amu = gm2calc.calculate_uncertainty_amu_2loop(model)
+    print(amu)
 except gm2calc.Error as e:
     print(e.what)
 
~~~
I get as output:
~~~
-276.5943268214615

Passed tests: [13/13]
~~~
This result is by no means equal to `3.2720909955357482*10.**(-11)`, but the test passes.

Reply:
So this test should have been registered as failing, but I mixed up the `True`/`False` values in the `if` statement in `testclose()`.  This test in particular was failing because of the lack of lambdas, but this is fixed now and the test gives correct results.  The other tests were failing because I mistakenly inputted the electron mass into the muon mass, or because of just a bit too high precision.  All tests should pass correctly now.

Reply:
Perfect, many thanks!